integrated_channels/canvas/client.py

```diff
@@ -90,9 +109,36 @@ def _delete(self, url, data):
             data (str): The json encoded payload to DELETE.
         """
         self._create_session()
-        response = self.session.delete(url, data=data)
+
+        integration_id = json.loads(
+            data.decode("utf-8")
+        )['course']['integration_id']
+        course_id = self._get_course_id_from_integration_id(integration_id)
+
+        url = '{}/api/v1/courses/{}'.format(
+            self.enterprise_configuration.canvas_base_url,
+            course_id,
+        )
+        response = self.session.delete(url, data='{"event":"delete"}')
```

Comment:
Apparently it's not good enough to simply make a `DELETE` request to canvas, you also have to supply this data because they've also included `concluding` a course within their `DELETE` endpoint... Maybe this is standard practice and I'm biased against this API at this point 😒

Reply:
yup, you have to ask for 'delete' or 'conclude' event. I doubt we ever want a conclude, so delete should be out default mode https://community.canvaslms.com/t5/Instructor-Guide/How-do-I-delete-a-course-as-an-instructor/ta-p/1181 also apparently it is possible to recover 'deleted' courses, so canvas itself looks like it is doing a soft delete. I think so.

Reply:
also check what the response of delete is in different situations... such as (1) course does not exist (2) course could not be deleted etc.

Reply:
Current iteration is raising Canvas' error response. However we may want to change this to be more clear in the future?